src/utils/jwt.ts

```diff
@@ -0,0 +1,52 @@
+// Cannot use endpoint since it does not allow any origin
+// Source: https://sharedeus.eus.attest.azure.net/certs
+// TODO: File an issue or figure out what we are doing wrong
+import certs from "./certs.json";
+import { b64utohex, RSAKey, X509 } from "jsrsasign";
+import { KJUR } from "jsrsasign";
+import { AttestationToken } from "./attestation-token";
+
+const ISSUER = "https://sharedeus.eus.attest.azure.net";
+
+// Keeping the function async since we will need it once the
+// we add the network call to get the certs
+export async function verifyToken(token: string): Promise<AttestationToken> {
+  const [header, payload] = token.split(".");
+
+  const headObj = KJUR.jws.JWS.readSafeJSONString(atob(header));
+
+  if (headObj == null) {
+    throw new Error("Invalid Token");
+  }
+
+  const key = getVerifyKey((headObj as { kid: string }).kid);
+
+  const isValid = KJUR.jws.JWS.verifyJWT(token, key, {
+    iss: [ISSUER],
+    alg: ["RS256"],
+    gracePeriod: 20 * 20 * 20 * 100 * 100
```

Comment:
We should add a flag to only have the grace period extended locally when developing.

Reply:
Is there any significance to the `20 * 20 * 20 * 100 * 100`?

(Was it supposed to be `60 * 60 * 24 * …`?)

Reply:
(nope, it is just a large number for testing with old tokens)